test/testRules.js

```diff
@@ -1674,5 +1675,193 @@ export const testRules = {
+                }),
+                new Result({
+                    rule,
+                    severity: "error",
+                    pathName: "x/y",
+                    locale: "de-DE",
+                    source: "Some source string with Some DNT term and Another DNT term in it.",
+                    id: "resource-dnt-test.dnt-missing-multiple",
+                    description: "A DNT term is missing in target string.",
+                    highlight: `Missing term: <e0>Another DNT term</e0>`,
+                })
+            ]
+        );
+        test.done();
+    },
+
+    testResourceDNTTermsOk: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceDNTTerms({
+            terms: [
+                "Some DNT term"
+            ]
+        });
+        test.ok(rule);
+
+        const subject = {
+            locale: "de-DE",
+            file: "x/y",
+            resource: new ResourceString({
+                key: "resource-dnt-test.dnt-ok",
+                sourceLocale: "en-US",
+                source: "Some source string with Some DNT term in it.",
+                targetLocale: "de-DE",
+                target: "Some target string with Some DNT term in it.",
+                pathName: "dnt-test.xliff",
+                state: "translated",
+            }),
+        };
+
+        const result = rule.match(subject);
+        test.deepEqual(result, []);
+        test.done();
+    },
+
+    testResourceDNTTermsLoadTermsNotProvided: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceDNTTerms({});
+        test.ok(rule);
+        test.deepEqual(rule.dntTerms, []);
+        
+        test.done();
+    },
+
+    testResourceDNTTermsLoadTermsFromConfig: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceDNTTerms({
+            terms: [
+                "Some DNT term",
+                "Another DNT term"
+            ]
+        });
+        test.ok(rule);
+
+        test.deepEqual(rule.dntTerms, [
+            "Some DNT term",
+            "Another DNT term"
+        ]);
+
+        test.done();
+    },
+    
+    testResourceDNTTermsLoadTermsFromJSONFile: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceDNTTerms({
+            termsFileType: "json",
+            termsFilePath: "./test/testfiles/dnt-test.json"
+        });
+        test.ok(rule);
+
+        test.deepEqual(rule.dntTerms, [
+            "Some DNT term",
+            "Another DNT term"
+        ]);
+
+        test.done();
+    },
+
+    testResourceDNTTermsLoadTermsFromTxtFile: function(test) {
+        test.expect(2);
+
+        const rule = new ResourceDNTTerms({
+            termsFileType: "txt",
+            termsFilePath: "./test/testfiles/dnt-test.txt"
+        });
+        test.ok(rule);
+
+        test.deepEqual(rule.dntTerms, [
+            "Some DNT term",
+            "Another DNT term"
+        ]);
+
+        test.done();
+    },
```

Comment:
Please add some tests to check for DNT terms in ResourceArray and ResourcePlural resources.